### PR TITLE
Add Plugin SolrAzure

### PR DESCRIPTION
**Pull request by Ashi-Jain1**
Signed-off-by: Ashi Jain <Ashi.Jain1@ibm.com>

### CODE REVIEW

**Comment by Ashi-Jain1** [comment]
Hi @Tal-Daniel  @PratikshaSonawane  @AlexaNahum ,
We have raised Apache Solr Azure PR, request to have a look.

Thanks

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+       $ cd /opt/solr-8.6.0
+       $ sudo bin/solr start -force 
+2.2.SolrCloud mode:  An index is distributed across multiple computers or even multiple server instances on one computer. Groups of documents here are called collections.
+ To Launch Solr in SolrCloud Mode:
+       $ cd /opt/solr-8.6.0
+       $ sudo bin/solr start -e cloud -force
+```
+
+## Adding Inbound port rule
+1. Navigate over to virtual machine.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:47]: Go to the virtual machine.

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+2. In the left-nav, select Networking.
+3. The rules will be displayed on right side of the portal.
+4. Select Inbound port rules ,then select Add.
+5. Add your port(on which solr is running, default is 8983) under Destination Port ranges.
+6. Click Add to create the rule.
+
+## Login to the Solr Dashboard
+To access the Solr admin panel, visit the hostname or IP address on port(solr is running):
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:55]: the port (on which Solr is running):

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    http://ip_address:port/solr/
+
+### Core Creation in Standalone mode
+```
+1.Create a new Solr core with the following command:
+  $ sudo bin/solr create -c core_name -force
+  For example: Core Created named as new_core:
+  $ sudo bin/solr create -c new_core -force
+2.Created core will reflect in the core drop-down menu on the solr admin console.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:64]: The created

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+```
+### Collection Creation in SolrCloud mode
+```
+1.Create a new Solr collection with the following command(having default shard and replica count):
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:68]: command with the default shard and replica count:

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+  $ sudo bin/solr create -c collection_name -force
+To create a collection having specified shard and replica count.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:70]: To create a collection with a customized shard and replica count ( still accurate?)

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+### Collection Creation in SolrCloud mode
+```
+1.Create a new Solr collection with the following command(having default shard and replica count):
+  $ sudo bin/solr create -c collection_name -force
+To create a collection having specified shard and replica count. 
+  $ sudo bin/solr create -c collection_name -s <count> -rf <count> -force
+  For example: Collection Created named as new_collection respectively.
+  $ sudo bin/solr create -c new_collection -force
+  $ sudo bin/solr create -c new_collection -s 1 -rf 2 -force
+2.Created collection will reflect in the collection drop-down menu on the solr admin console.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:75]: The created

Solr

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+```    
+
+## View the Apache Solr logs
+To configure logging, visit Solr Dashboard. Under Java Properties , search for solr.​log.​dir<br>
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:80]: the Solr dashboard. 

Under Java properties,

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+To view logs goto Location:<br>
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:81]: logs, go to

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+To install Filebeat on your system, follow the steps in this topic:
+    https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-installation-configuration.html#installation
+
+## b. Filebeat configuration:
+
+To use Logstash to perform additional processing on the data collected by Filebeat, we need to configure Filebeat to use Logstash. To do this, modify the filebeat.yml file which you can find inside the folder where filebeat is installed. Follow these instructions for finding the installation directory:
+    https://www.elastic.co/guide/en/beats/filebeat/current/directory-layout.html
+
+In addition, events are configured with the include_lines, exclude_lines and add_tags. You can learn more about Filebeat Configuration [here](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-log.html#:~:text=include_lines%20edit%20A%20list%20of%20regular%20expressions%20to,the%20list.%20By%20default%2C%20all%20lines%20are%20exported.).
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:106]: exclude_lines,

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+              - /opt/solr-8.6.0/example/cloud/node2/logs/solr.log
+              - /opt/solr-8.6.0/example/cloud/node3/logs/solr.log
+              - /opt/solr-8.6.0/example/cloud/node4/logs/solr.log
+			  
+	• To include logs that are the result of RequestHandler, LogUpdateProcessorFactory, HttpSolrCall, and Exceptions in Solr.
+	
+            include_lines: ['o.a.s.u.p.LogUpdateProcessorFactory','o.a.s.c.S.Request','o.a.s.h.RequestHandlerBase org.apache.solr.common.SolrException','o.a.s.s.HttpSolrCall','o.a.s.c.a.c.OverseerCollectionMessageHandler','o.a.s.c.s.i.s.ExceptionStream','o.a.s.h.SQLHandler','o.a.s.h.e.ExportWriter','o.a.s.h.StreamHandler']
+            exclude_lines: ['status=400']
+
+    • To process multi-line audit events, add the settings in same inputs section.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:129]: the same

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+    • Enable Logstash output by uncommenting the Logstash section. For more information, see https://www.elastic.co/guide/en/beats/filebeat/current/logstash-output.html#logstash-output
+
+    For example:
+            output.logstash:
+	      hosts: ["20.115.120.146:5085"]
+
+    • The hosts option specifies the Logstash server and the port (5001) where Logstash is configured to listen for incoming Beats connections.
+
+    •You can set any port number except 5044, 5141, and 5000 (as these are currently reserved in Guardium v11.3 and v11.4 ).
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:154]: is this still true?

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+```
+
+3. To learn how to start FileBeat, see https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-installation-configuration.html#start
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:157]: Filebeat (I think that one is a previous mistake of mine, sorry)

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+
+
+## Configuring the ApacheSolr filter in Guardium
+The Guardium universal connector is the Guardium entry point for native audit logs. The Guardium universal connector identifies and parses the received events, and converts them to a standard Guardium format. The output of the Guardium universal connector is forwarded to the Guardium sniffer on the collector, for policy and auditing enforcements. Configure Guardium to read the native audit logs by customizing the solr template.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:161]: Solr

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+
+## Before you begin
+* You must have Log Full Details policy enabled on the collector. The detailed steps can be found in step #4 on [this page](https://www.ibm.com/docs/en/guardium/11.4?topic=dpi-installing-testing-filter-input-plug-in-staging-guardium-system).
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:164]: have the

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+* You must have permission for the S-Tap Management role. The admin user includes this role by default.
+* Download the [guardium_logstash-offline-plugin-apache-solr-azure.zip](ApacheSolrOverFilebeatPackage/guardium_logstash-offline-plugin-apache-solr-azure.zip) plug-in.
+
+## Procedure
+1. On the collector, go to Setup > Tools and Views > Configure Universal Connector.
+2. First Enable the Universal Guardium connector, if it is Disabled already.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:170]: First, enable

if it is currently disabled.

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+3. Click Upload File and select the offline [guardium_logstash-offline-plugin-apache-solr-azure.zip](ApacheSolrOverFilebeatPackage/guardium_logstash-offline-plugin-apache-solr-azure.zip) plug-in. After it is uploaded, click OK.
+4. Click the Plus sign to open the Connector Configuration dialog box.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:172]: Plus icon

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+5. Type a name in the Connector name field.
+6. Update the input section to add the details from the [solrazure.conf](solrazure.conf) file's input part, omitting the keyword "input{" at the beginning and its corresponding "}" at the end.
+7. Update the filter section to add the details from the [solrazure.conf](solrazure.conf) file's filter part, omitting the keyword "filter{" at the beginning and its corresponding "}" at the end.
+8. The 'type' fields should match in the input and filter configuration sections. This field should be unique for every individual connector added.
+9. The tag added in the filebeat.yml file should match the "[tags]" specified in the filter part.
+10. Click Save. Guardium validates the new connector, and enables the universal connector if it was disabled. After it is validated, it appears on the Configure Universal Connector page.
+
+## Limitations    
+1. Following important fields couldn't be mapped with ApacheSolr qtp logs.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:181]: The following

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 18]
```diff
@@ -0,0 +1,189 @@
...
+     - SourceProgram : field is left blank since this information is not embedded in the messages pulled from Azure.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:182]: blank, as this (for consistency)

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+     - clientIP : field is populated with 0.0.0.0, as this information is not embedded in the messages pulled from Azure.
+     - OS User         : Not available with logs
+     - Client HostName : Not available with logs
+     - dbUser          : Not available with logs
+     - LOGIN_FAILED    : Not available with logs
+
+2. While launching Solr in SolrCloud mode, multiple logs will be generated for single query execution as a call to shard(In SolrCloud, a logical partition of a single Collection) and replica(A Core that acts as a physical copy of a Shard in a SolrCloud Collection).
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:189]: core. shard. (lower-case)

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/gradlew [hunk 20]
```diff
@@ -0,0 +1,185 @@
...
+        warn "Could not query maximum file descriptor limit: $MAX_FD_LIMIT"
+    fi
+fi
+
+# For Darwin, add options to specify how the application appears in the dock
+if $darwin; then
+    GRADLE_OPTS="$GRADLE_OPTS \"-Xdock:name=$APP_NAME\" \"-Xdock:icon=$APP_HOME/media/gradle.icns\""
+fi
+
+# For Cygwin or MSYS, switch paths to Windows format before running java
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/gradlew:129]: Java

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/src/main/java/com/ibm/guardium/apachesolrdb/CommonUtils.java [hunk 21]
```diff
@@ -0,0 +1,101 @@
+/*
+� Copyright IBM Corp. 2021, 2022 All rights reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+package com.ibm.guardium.apachesolrdb;
+
+/**
+ * this class contains all the utility methods which contains the common
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/src/main/java/com/ibm/guardium/apachesolrdb/CommonUtils.java:8]: contain?

File: filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md [hunk 22]
```diff
@@ -0,0 +1,189 @@
+# Apache Solr Azure-Guardium Logstash filter plug-in
+
```
Comment (PratikshaSonawane) [inline filter-plugin/logstash-filter-azure-apachesolr-guardium/README.md:2]: It seems this README is not in the format that template is shared for Filebeat.

**Review by PratikshaSonawane — APPROVED** [review]

**Pull request merged**